source/Plugins.h

```diff
@@ -57,12 +79,24 @@ class Plugins {
 	// Returns true if any plugin enabled or disabled setting has changed since
 	// launched via user preferences.
 	static bool HasChanged();
+	// Returns true if there is active install/update activity;
+	static bool IsInBackground();
 
 	// Returns the list of plugins that have been identified by the game.
 	static const Set<Plugin> &Get();
 
 	// Toggles enabling or disabling a plugin for the next game restart.
 	static void TogglePlugin(const std::string &name);
+
+	// Install or update or delete a plugin.
+	static std::future<void> Install(const InstallData &installData, bool guarded = false);
+	static std::future<void> Update(const InstallData &installData);
+	static void DeletePlugin(const InstallData &installData);
+
+	static bool Download(std::string url, std::string location);
+
+private:
+	static bool ExtractZIP(std::string filename, std::string destination, std::string expectedName);
```

Comment:
This should be a anonymous function in Plugins.cpp

Reply:
did that